app/lib/actions/remote_execution/run_proxy_command.rb

```diff
@@ -39,18 +39,56 @@ def exit_status
       end
 
       def live_output
-        output = self.output[:proxy_output].present? ? self.output[:proxy_output] : fetch_output['output']
-        output_records(output)
+        records = connection_messages
```

Comment:
The assumption is the connection error will always happen sooner, than other execution output: in other words, once the connection is established, no other connection errors are expected to occur. But I can make a final sort, just in case